concurrent_test.go

```diff
@@ -738,3 +754,135 @@ func validateSequential(rs historyRecords) error {
 
 	return nil
 }
+
+func TestConcurrentRepeatableRead(t *testing.T) {
+	t.Log("Preparing db.")
+	var (
+		bucket = []byte("data")
+		key    = []byte("mykey")
+
+		option = &bolt.Options{PageSize: 128}
+	)
+
+	db := mustCreateDB(t, option)
+	defer func() {
+		db.Close()
+	}()
```

Comment:
The suggested change doesn't work, because the db is reopened later. The old db is already closed.

Reply:
Note that the execution of a deferred function is deferred to the moment the surrounding function returns, but the function value and parameters to the call are evaluated as usual and saved anew.

Reply:
Can you add this information as comment? It's not obvious at first sight. `var db` is being re-used, maybe if you can rename them as `db1` and `db2` to avoid re-using the variable. Or better, extract the whole preparing section into another function to completely avoid this problem. Does it make sense?

Reply:
It seems that renaming to `db1` vs `db2` can't resolve this, nor extracting the preparing section into another function, as long as we need to reopen the db and use deferred function to close the db.

Added a comment anyway.